lua/randomat2/cl_events/cl_poker.lua

```diff
@@ -0,0 +1,498 @@
+                    newCard.Kept = true
+                end
+            end
+        end
+
+        for _, card in ipairs(newHand) do
+            if not card.Kept then
+                newCards = newCards .. CardRankToName(card.Rank) .. " of " .. CardSuitToName(card.Suit) .. "\n"
+            end
+        end
+    end
+
+    self.Hand = newHand
+    self.PokerHand:SetHand(newHand)
+
+    if isSecondDeal then
+        if newCards ~= "" then
+            self.PokerMain:TemporaryMessage("Your new cards:\n" .. newCards)
+        else
+            self.PokerMain:TemporaryMessage("No new cards given")
+        end
+    end
+end
+
+function EVENT:StartBetting(ply, timeToBet)
+    if ply == self.Self then
+        self.PokerMain:TemporaryMessage("Your turn to bet!")
+        self.PokerMain:SetTimer(timeToBet)
+        self.PokerControls:EnableBetting()
+
+        if ConVars.EnableRoundStateAudioCues:GetBool() then
+            surface.PlaySound("poker/chips.ogg")
+        end
+    else
+        self.PokerMain:TemporaryMessage(ply:Nick() .."'s turn to bet!")
+    end
+
+    self.PokerPlayers:SetPlayerAsBetting(ply)
+    self.PokerHand:SetCanDiscard(false)
+end
+
+function EVENT:RegisterBet(ply, betType, betAmount, plySteamId)
+    if ply == LocalPlayer() then
+        if betType == BettingStatus.FOLD then
+            self.PokerMain:SetSelfFolded()
+            self:EnableClose()
+        elseif self.PokerControls and self.PokerControls:IsValid() then
+            self.PokerControls:SetCurrentBet(betAmount or self.PokerControls.CurrentRaise)
+            self.PokerControls:DisableBetting()
+        end
+    
+        self.PokerMain:SetTimer(0)
+    else
+        self.PokerPlayers:SetPlayerBet(ply, betType, betAmount or self.PokerControls:GetCurrentRaise(), plySteamId)
+
+        if betType == BettingStatus.RAISE and self.PokerControls and self.PokerControls:IsValid() and betAmount then
+            self.PokerControls:SetCurrentRaise(betAmount)
+        end
+    end
+
+    if IsAllIn(betAmount) then
+        self.PokerControls:DisableRaising()
+    end
+end
+
+function EVENT:EndBetting()
+    self.PokerMain:TemporaryMessage("Betting completed!")
+    self.PokerPlayers:SetPlayerAsBetting()
+    self.PokerControls:DisableBetting()
+end
+
+function EVENT:BeginDiscarding(timeToDiscard)
+    self.PokerMain:TemporaryMessage("Now, discard up to three cards!")
+    self.PokerMain:SetTimer(timeToDiscard)
+    self.PokerHand:SetCanDiscard(true)
+    self.PokerPlayers:ResetPlayerActions()
+
+    if ConVars.EnableRoundStateAudioCues:GetBool() then
+        surface.PlaySound("poker/shuffle.ogg")
+    end
+end
+
+function EVENT:EndDiscard()
+    self.PokerMain:SetTimer(0)
+end
+
+function EVENT:RegisterWinner(winner, hand)
+    if winner then
+        if winner == LocalPlayer() then
+            self.PokerMain:PermanentMessage("You win! Getting your bonus health now!")
+
+            if ConVars.EnableYogsification:GetBool() and self.ShouldPlayStartSound then
```

Comment:
Forgot to include ConVars.EnableRoundStateAudioCues:GetBool() check here and line 258